atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/cleaner/AsyncPuncherTest.java

```diff
@@ -74,27 +93,43 @@ public void delegatesInitializationCheck() {
 
     @Test
     public void testPuncherDurability() throws Exception {
-        Long stored = timestampService.getFreshTimestamp();
-        asyncPuncher.punch(stored);
-        Long retrieved = Long.MIN_VALUE;
-        for (int i = 0; i < MAX_INTERVALS_TO_WAIT && retrieved < stored; i++) {
-            Thread.sleep(ASYNC_PUNCHER_INTERVAL);
-            retrieved = asyncPuncher.getTimestampSupplier().get();
-        }
-        assertEquals(stored, retrieved);
+        long stored = timestampService.getFreshTimestamp();
+        asyncPuncherNonSeeded.punch(stored);
+        checkExpectedValue(asyncPuncherNonSeeded, stored);
     }
 
+
     @Test
     public void testPuncherTimestampLessThanFreshTimestamp() throws Exception {
         Long stored = timestampService.getFreshTimestamp();
-        asyncPuncher.punch(stored);
+        asyncPuncherNonSeeded.punch(stored);
         Long retrieved = Long.MIN_VALUE;
         for (int i = 0; i < MAX_INTERVALS_TO_WAIT && retrieved < stored; i++) {
             Thread.sleep(ASYNC_PUNCHER_INTERVAL);
-            retrieved = asyncPuncher.getTimestampSupplier().get();
+            retrieved = asyncPuncherNonSeeded.getTimestampSupplier().get();
         }
         long freshTimestamp = timestampService.getFreshTimestamp();
         assertTrue(retrieved < freshTimestamp);
     }
 
+    @Test
+    public void testPuncherStartUpSeed() throws Exception{
+        checkExpectedValue(asyncPuncherSeeded, PUNCHER_SEED);
+    }
+
+    @Test
+    public void testSeededPuncherUpdate() throws Exception {
+        asyncPuncherSeeded.punch(PUNCHER_UPDATE_TIMESTAMP);
+        checkExpectedValue(asyncPuncherSeeded, PUNCHER_UPDATE_TIMESTAMP);
+    }
+
+    private void checkExpectedValue(Puncher puncher, Long expected) throws InterruptedException {
+        Long retrieved = Long.MIN_VALUE;
```

Comment:
Yeah, I guess this is consistent with the style of the other tests. I'd probably write it with the primitive (`long`) form, but this should be fine

Reply:
Updated to long.